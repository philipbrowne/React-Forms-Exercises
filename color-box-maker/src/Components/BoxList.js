import React, { useState } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import './BoxList.css';
import { v4 as uuid } from 'uuid';

const BoxList = () => {
  const INITIAL_STATE = [
    {
      color: 'blue',
      width: 500,
      height: 300,
      id: uuid(),
    },
    {
      color: 'red',
      width: 200,
      height: 400,
      id: uuid(),
    },
  ];
  const [boxes, setBoxes] = useState(INITIAL_STATE);
  const addBox = (newBox) => {
    setBoxes((boxes) => [...boxes, { ...newBox, id: uuid() }]);
  };
  const remove = (box) => {
    setBoxes(boxes.filter((b) => b !== box));
  };
  return (
    <div className="BoxList">
      <h1>Colored Boxes!</h1>
      <NewBoxForm addBox={addBox} />
      <div className="BoxList-container">
        {boxes.map((b) => (
          <Box
            color={b.color}
            width={b.width}
            height={b.height}
            key={b.id}
            remove={(evt) => remove(b)}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
