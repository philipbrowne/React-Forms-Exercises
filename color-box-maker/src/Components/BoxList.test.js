import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';

it('renders without crashing', () => {
  render(<BoxList />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it('should add new item', () => {
  const { queryByText, getByLabelText, getAllByRole } = render(<BoxList />);
  expect(queryByText('Width: 300 pixels')).not.toBeInTheDocument();
  const widthInput = getByLabelText('Width: 0 pixels');
  const heightInput = getByLabelText('Height: 0 pixels');
  const btn = queryByText('Add Box');
  fireEvent.change(widthInput, { target: { value: '250' } });
  fireEvent.change(heightInput, { target: { value: '250' } });
  fireEvent.click(btn);
  const boxes = getAllByRole('figure');
  expect(boxes.length).toBe(3);
});
