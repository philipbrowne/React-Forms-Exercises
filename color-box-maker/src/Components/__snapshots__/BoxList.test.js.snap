// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h1>
      Colored Boxes!
    </h1>
    <div>
      <form>
        <div
          class="NewBoxForm-inputs"
        >
          <div
            class="NewBoxForm-input"
          >
            <label
              class="NewBoxForm-label"
              for="color"
            >
              Color
            </label>
            <input
              id="color"
              name="color"
              required=""
              type="color"
              value="#000000"
            />
          </div>
          <div
            class="NewBoxForm-input"
          >
            <label
              class="NewBoxForm-label"
              for="width"
            >
              Width: 
              <span>
                0 pixels
              </span>
            </label>
            <input
              id="width"
              max="300"
              min="100"
              name="width"
              required=""
              type="number"
              value="0"
            />
          </div>
          <div
            class="NewBoxForm-input"
          >
            <label
              class="NewBoxForm-label"
              for="height"
            >
              Height: 
              <span>
                0 pixels
              </span>
            </label>
            <input
              id="height"
              max="300"
              min="100"
              name="height"
              required=""
              type="number"
              value="0"
            />
          </div>
        </div>
        <div
          class="NewBoxForm-submit"
        >
          <button>
            Add Box
          </button>
        </div>
      </form>
    </div>
    <div
      class="BoxList-container"
    >
      <div
        class="Box"
        role="figure"
        style="background-color: blue; width: 500px; height: 300px; margin: 1rem;"
      >
        <button
          class="Box-btn"
        >
          X
        </button>
      </div>
      <div
        class="Box"
        role="figure"
        style="background-color: red; width: 200px; height: 400px; margin: 1rem;"
      >
        <button
          class="Box-btn"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
